lib/configproxy.js

```diff
@@ -379,19 +379,23 @@ class ConfigurableProxy extends EventEmitter {
   }
 
   targetForReq(req) {
-    var metricsTimerEnd = this.metrics.findTargetForReqSummary.startTimer();
-    // return proxy target for a given url path
-    var basePath = this.hostRouting ? "/" + parseHost(req) : "";
-    var path = basePath + decodeURIComponent(URL.parse(req.url).pathname);
-
-    return this._routes.getTarget(path).then(function (route) {
-      metricsTimerEnd();
-      if (route) {
-        return {
-          prefix: route.prefix,
-          target: route.data.target,
-        };
-      }
+    return new Promise((resolve, reject) => {
```

Comment:
Noting for review:

wrapping this in a promise ensures that targetForReq always returns a Promise, even if `routes.getTarget` raises directly (as opposed to returning a failing promise).